### DISCUSSION TITLE
How to test Fedosov Coarse-Grained Spectrin-Link membarne model?

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#45] [General discussion]
Hello,

I know Fedosov's method is already implemented in HemoCell but I cannot seem to find where it is or how I can enable it. If it is available can you please tell me how to set it up?

Much appreciated,
Mojtaba

**Comment by gzavo** [comment]
It was implemented, but we had lots of numerical issues with it at moderate and high shear rates, so we ended up using our method all the time, and eventually we dropped support for it.
It is probably not a big challenge to reimplement it if you need it for some reason, look for the "HemoCell/mechanics" folder. Duplicate the "rbcHighOrderModel" cpp and header. The function "ParticleMechanics" contains the whole mechanical model, that is the one needs to be reimplemented with the Fedosov model. Currently it looks a bit complicated, because of some optimisation work that fused certain loops and reorganised things to make the code run faster, but you can retain a form closer to the equations.

**Reply by mojtabaAmiraslanpour** [reply]
Did you have any ideas where those numerical issues originated?

**Reply by gzavo** [reply]
As I recall most problems arose around the bending force. It was accurate (as far as the validations could confirm) for small deformations, but under higher shear rates it allowed extreme bends (folding back completely along one edge of neighbouring surface triangles). This also led to issues with immersed boundary. My guess is that in their DPD-based model this issue never caused a problem due to the type of discretization. I.e., you can't fully fold flat a set of spheres. Anyway, we made the assumption that under very large deformations the cytosceleton is also getting involved, not just the membrane, so infinitely high curvature (flat fold) shouldn't happen.
